### DISCUSSION TITLE
react组件的冒泡

### DISCUSSION THREAD

**Discussion by robertpanvip** [antvis/X6#3613] [Q&A]
如何阻止自定义的react组件元素的冒泡事件

**Comment by lloydzhou** [comment]
## 举个例子：
比如react自定义组件包含input或者button的时候，点击button会触发selection plugin的交互，组件直接被选中。

## 讨论：
1. https://github.com/antvis/X6/pull/2676 之前有提过一个pr，针对react组件，处理了onMouseDown和onMouseUp事件，过滤不必要的事件冒泡。　　但是，中间2.x大版本更新整体迁移代码的时候，漏掉了中间的部分pr，所以现在问题还在。
2. 最近写了一个新的[x6-html-shape](https://github.com/lloydzhou/x6-html-shape/blob/master/src/index.ts#L70)，这个库支持使用官方的foreignObject以及普通的html元素自定义节点。支持react/vue/svelet等不同框架不同版本进行集成
![](https://user-images.githubusercontent.com/1826685/238824053-1ea8cbc4-49f1-478b-a0b7-228eea993f88.png)
